packages/schematics/angular/service-worker/files/ngsw-config.json.template

```diff
@@ -11,7 +11,8 @@
           "/index.html",
           "/manifest.webmanifest",
           "/*.css",
-          "/*.js"
+          "/*.js",
+          "!/*-es5*.js"
```

Comment:
If we are doing it, then we should also lazily cache the `/*-es5*.js` bundles (as shown in https://github.com/angular/angular/issues/31256#issuecomment-506507021).
There is no overhead for es2015 supporting browsers.

Reply:
Good point

Reply:
> It would be even better imo if we could adapt the ngsw-config.json template to the TS config target. Is it possible to do that, @alan-agius4?

A different tsconfig can be specified for each configuration. An application can various tsconfig, though this is probably not the norm. 

That said, differential loading though is not only controlled via the `target` specified in the tsconfig, but also the result of the browserslists query.

@gkalpak, what would we the value to generate a different `ngsw-config.json`?

Reply:
In version 10 differential loading is on opt-in basis.

Reply:
I actually have some concerns about adding this. 

For the reasons, that in version 11 we might want to change the generated bundle names not to have the syntax target as part of the filename.

See for more context: https://github.com/angular/angular-cli/pull/17630#pullrequestreview-405834936

This would means that by adding this, we'd need to do a migration to migrate existing projects and replace the negate pattern.

Reply:
So, what is the plan for differentiating es5 bundles vs es20XX bundles in v11?

Reply:
It hasn't been decided yet.

There are two 2 main option here, either leave the script target in the filename or change the filename not to contain it and use generic terms, like 'modern', legacy' etc...

Reply:
> This would means that by adding this, we'd need to do a migration to migrate existing projects and replace the negate pattern.

v11 is still many months away though, so while I do agree that not having `es2015` in the bundle names is more future proof I think there is some value in introducing that logic now & doing a migration when v11 lands, especially as double (pre)fetching by the Angular Service Worker can be hard to detect when not looking for it.

Having said that I'll likely won't be the one writing the v11 migration code so I completely understand if you want to postpone that for now, and I'll try not to forget to update my `ngsw-config.json` for v11 :).